internal/service/indexer/transfer.go

```diff
@@ -0,0 +1,272 @@
+		if err != nil {
+			return errors.Wrap(err, "failed to update leaf in merkle tree")
+		}
+	}
+
+	err = i.ContractsQ.FilterByAddresses(event.Raw.Address.Hex()).Update(data.ContractToUpdate{
+		Block: &blockNumber,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to update last handled block")
+	}
+
+	err = i.publish(IssuerContract, mTree.Root())
+	if err != nil {
+		return errors.Wrap(err, "failed to publish")
+	}
+
+	i.log.WithField("address", event.Raw.Address.Hex()).Debugf("finish handling mint event")
+	return nil
+}
+
+func (i *Indexer) completelyDeleteKey(mTree *merkletree.MerkleTree, event *contracts.TokenContractTransfer, treeStorage *postgres.Storage) error {
+	err := mTree.Delete(i.ctx, event.To.Big())
+	if err != nil {
+		return errors.Wrap(err, "failed to delete address from merkle tree")
+	}
+
+	dump, err := mTree.DumpLeafs(i.ctx, mTree.Root())
+	if err != nil {
+		return errors.Wrap(err, "failed to make dump of merkle tree")
+	}
+
+	err = treeStorage.DeleteMTree(i.ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete mtree")
+	}
+
+	contract, err := i.ContractsQ.FilterByAddresses(event.Raw.Address.Hex()).Get()
+	if err != nil {
+		return errors.Wrap(err, "failed to get contract")
+	}
+	if contract == nil {
+		return errors.New("no contract was found")
+	}
+
+	treeStorage = postgres.NewPGDBStorage(i.cfg.DB(), contract.Id)
+	newMTree, err := merkletree.NewMerkleTree(i.ctx, treeStorage, 100)
+	if err != nil {
+		return errors.Wrap(err, "failed to create merkle tree")
+	}
+
+	err = newMTree.ImportDumpedLeafs(i.ctx, dump)
+	if err != nil {
+		return errors.Wrap(err, "failed to import dumped leafs")
+	}
+
+	return nil
+}
+
+func (i *Indexer) publish(name string, root *merkletree.Hash) error {
+	client, err := ethclient.Dial(i.cfg.Infura().Sepolia + i.cfg.Infura().Key)
+	if err != nil {
+		return errors.Wrap(err, "failed to make dial connect sepolia")
+	}
+	err = i.sendUpdates(client, name, root)
+	if err != nil {
+		return errors.Wrap(err, "failed to publish in sepolia")
+	}
+
+	client, err = ethclient.Dial(i.cfg.Infura().Polygon + i.cfg.Infura().Key)
+	if err != nil {
+		return errors.Wrap(err, "failed to make dial connect polygon")
+	}
+	err = i.sendUpdates(client, name, root)
+	if err != nil {
+		return errors.Wrap(err, "failed to publish in polygon")
+	}
+
+	//TODO: publish to Q Testnet
+
+	return nil
+}
+
+func (i *Indexer) sendUpdates(client *ethclient.Client, name string, root *merkletree.Hash) error {
+	auth, err := i.getAuth(client)
+	if err != nil {
+		return errors.Wrap(err, "failed to get auth options")
+	}
+
+	var course [32]byte
+	copy(course[:], name)
+
+	certIntegrator, err := contracts.NewCertIntegratorContract(common.HexToAddress(i.cfg.CertificatesIntegrator().Ethereum), client)
+	if err != nil {
+		return errors.Wrap(err, "failed to create new cert integrator contract")
+	}
+
+	_, err = certIntegrator.UpdateCourseState(auth, [][32]byte{course}, [][32]byte{*root})
```

Comment:
It would be best if you waited while a TX will be mined. What if it will revert? Create a runner where you will put the TX, and it will wait while it will be mined.